### ISSUE TITLE
Problem with bitbucket issue

### ISSUE THREAD

**Issue by dartess** [vordgi/nimpl-classnames-minifier#81]
Hi!

When the minifier detects changes in the configuration, it cleans the build folder (by default this is `.next`).

In Vercel and localhost it is ok. But our dev pipeline works like this:

1. on the first machine (bitbucket pipeline) the build occurs.
2. the build is sent to the second machine (hosting).
3. on the second machine (hosting) the start occurs.

At the moment of start, the plugin deletes the build folder and we get this:
![image](https://github.com/vordgi/nimpl-classnames-minifier/assets/15047511/afabbde6-50bb-4715-b408-67f1f7b7b93b)

I'm not sure if the plugin does useful work in runtime, or only in buildtime? If only in buildtime, is it possible to check this somehow and skip the work when starting the production server (`next start`)?

**Comment by vordgi** [comment]
Hello!

Unfortunately, within the package setup function, package can't determine if it's the build time or the application run. This can only be done by the `phase` arg, which is passed by next.js

Use the `disabled` key, as described on the [configuration page](https://nimpl.tech/classnames-minifier/configuration)

```js
const { default: classnamesMinifier } = require('@nimpl/classnames-minifier');

module.exports = (phase) => classnamesMinifier({
  prefix: '_',
  reservedNames: ['_en', '_de'],
  disabled: phase === PHASE_PRODUCTION_SERVER || process.env.NODE_ENV === 'development',
})(nextConfig);
```

Also, in the near future, I plan to add a separate option specifically for disabling auto-deletion, but in your case, the current `disabled` option will be sufficient.

Thank you for the issue, I will think how to improve package api and documentation to make this moment clearer

**Comment by dartess** [comment]
Thanks, that sounds like a solution!

> Although the question here is why the package thought there were changes in the configuration.

Unfortunately I cannot provide details as I do not have full access to all the necessary resources. I was only provided with a screenshot with this problem.

> Can you send the package configuration, please?

Are we talking about the plugin settings? They are primitive:

```js
withClassnamesMinifier({
  disabled: process.env.NODE_ENV === 'development',
  prefix: '_',
})(nextConfig);
```

**Comment by vordgi** [comment]
Yes, about them. Strange. For some reason, the package in the second machine thinks that these options have changed. I will try to figure it out as soon as possible.

Is the same version of the package being used in both of machines, is the .next/cache folder being cloned to the second machine?

**Comment by dartess** [comment]
I think it is completely cloned.

Yes, it was probably worth mentioning: it started after the migration from `next-classnames-minifier` to `@nimpl/classnames-minifier`.

**Comment by vordgi** [comment]
I released version 3.0.2. It doesn't have a fix, but cache deletion messages had been improved.

I would be grateful if you will run the build with the new version and send what message is logged in the terminal.

**Comment by dartess** [comment]
Unfortunately, I don't have the ability to update dependencies outside of the schedule. If I encounter any problems next time I do a scheduled update, I'll be happy to report them.

I think the current [configuration description in the documentation](https://nimpl.tech/classnames-minifier/configuration) is enough to avoid a problem like mine and this issue can be closed.

BTW about docs:
a) `disabled: phase === PHASE_DEVELOPMENT_SERVER || phase === PHASE_PRODUCTION_SERVER` seems to fit too, and also looks more uniform (instead of using `process.env.NODE_ENV`).
b) I had never encountered such constants before, and I couldn’t even quickly find documentation on the `phase` argument. In the connection example, you can add import from `next/constants` for inexperienced users.

**Comment by vordgi** [comment]
Thanks for the suggestions! Made changes to the documentation.

And thank you for writing about the problem, it's very valuable!
